warehouse/query-core/src/main/java/org/apache/commons/jexl2/parser/JexlNodes.java

```diff
@@ -199,4 +199,37 @@ public static ASTNumberLiteral literal(Number n) {
         l.literal = n;
         return l;
     }
+    
+    /**
+     * Remove childToRemove from parent, updating all references on both the parent and childToRemove to be consistent
+     * 
+     * @param parent
+     *            the parent to remove from
+     * @param childToRemove
+     *            the child to remove from parent
+     * @return true if childToRemove was successfully found and removed from parent, false otherwise
+     */
+    public static boolean removeFromParent(JexlNode parent, JexlNode childToRemove) {
+        boolean found = false;
+        JexlNode[] children = new JexlNode[parent.jjtGetNumChildren() - 1];
+        int childIndex = 0;
+        for (int i = 0; i < parent.jjtGetNumChildren(); i++) {
+            JexlNode child = parent.jjtGetChild(i);
+            if (!child.equals(childToRemove)) {
+                children[childIndex++] = child;
```

Comment:
If we do not find the child, then I think we will hit an array out of bounds exception here

Reply:
good catch